packages/core/src/browser/shell/tab-bar-decorator.ts

```diff
@@ -0,0 +1,88 @@
+/********************************************************************************
+ * Copyright (C) 2019 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { inject, injectable, named, postConstruct } from 'inversify';
+import { Event, Emitter, Disposable, DisposableCollection, ContributionProvider } from '../../common';
+import { Title, Widget } from '@phosphor/widgets';
+import { Decoration } from '../tree/decoration';
```

Comment:
Tree specific api should be used in shell api. It can be other way around, tree can depend on some icon decorations.

Also:
- it is a big breaking change to move tree decorations make sure it is highlighted in CHANGELOG
- `Decoration` is too generic and bad name, we should carefully tab bar decoration from tree decoration

Reply:
For the naming, how about `NodeDecoration`, since it is used both by tree nodes and tabbar items (nodes)?

Reply:
For the naming, how about `NodeDecoration`, since it is used by both tree nodes and tabbar items (nodes)?

Reply:
What does `Node` stand for?

Reply:
`Node` can refer to `tree node`, `tabbar tab`, or any other item that has the similar (icon + label) UI.

Reply:
Just had a discussion with Vince. We thought it is better to name it `WidgetDecoration`, since `Node` can be sort of confusing. How does that sound to you? ;)

Reply:
yes, i am also struggling to come up with better naming, `Widget` sounds better, I thought maybe `HTMLElementDecoration`, but still not sure